internal/api/impl/auth/auth.go

```diff
@@ -141,3 +143,52 @@ func (e *endpoint) logout(ctx echo.Context) error {
 
 	return ctx.Redirect(302, "/")
 }
+
+// withOAuthErrorMdw applies to the handler a middleware that the error is well managed in oauth2.0 context.
+// This will transform any oauth2.0 retrieve token error into a bad request error, as suggested in the specs:
+// https://www.rfc-editor.org/rfc/rfc6749#section-5.2
+// https://www.rfc-editor.org/rfc/rfc8628#section-3.5
+// Otherwise, and only if the error is not an explicit Perses or HTTP error, we send back an opaque InternalError.
+func withOAuthErrorMdw(handler echo.HandlerFunc) echo.HandlerFunc {
+	return func(ctx echo.Context) error {
```

Comment:
```suggestion
	return func(ctx echo.Context) error {
		err := next(ctx)
		if err == nil {
			return nil
		}
		
		// This error is the token error that could come from the oauth2 library
		var retrieveError *oauth2.RetrieveError
		if errors.As(err, &retrieveError) {
			return ctx.JSON(http.StatusBadRequest, api.OAuthError{
				ErrorCode:        retrieveError.ErrorCode,
				ErrorDescription: retrieveError.ErrorDescription,
			})
		}
		// This error is the oidc error that could come from the oidc library
		var oidcError *oidc.Error
		if errors.As(err, &oidcError) {
			return ctx.JSON(http.StatusBadRequest, api.OAuthError{
				ErrorCode:        string(oidcError.ErrorType),
				ErrorDescription: oidcError.Description,
			})
		}
		
		// for all other error, we don't need to manage it as it will be handled by the next middleware `HandleError`.
		return err
}
```